File: 11-DYNAMIC-PROGRAMMING/PHP/house-robber.php
<?php

// https://leetcode.com/problems/house-robber/

class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function rob(array $nums): int
    {
        $rob1 = 0;
        $rob2 = 0;
        foreach ($nums as $num) {
            $tmp = max($rob1+$num, $rob2);
            $rob1 = $rob2;
            $rob2 = $tmp;
        }

        return $rob2;
    }
}

File: 11-DYNAMIC-PROGRAMMING/Python/house-robber.py
from typing import List

# https://leetcode.com/problems/house-robber/

class Solution:
    def rob(self, nums: List[int]) -> int:
        rob1, rob2 = 0, 0
        for num in nums:
            rob1, rob2 = rob2, max(rob1+num, rob2)

        return rob2